### PR TITLE
Formalises proof by Bernhard Rienke that 1648 implies 206 does not hold

### PR DESCRIPTION
**Pull request by Shreyas4991**
See the thread for the proof description and formalisation:
https://leanprover.zulipchat.com/#narrow/stream/458659-Equational/topic/1648.20!.3D.3E.20206/near/476775769

### CODE REVIEW

File: equational_theories/ManuallyProved/Equation1648.lean [hunk 1]
```diff
@@ -19,3 +20,19 @@ theorem Equation1648_not_implies_Equation206 : ∃ (G: Type) (_: Magma G), Equat
...
+  Equation1648 G ∧ ¬ Equation151 G := by
+  let instMagmaInt : Magma ℤ := {
+    op := fun x y => if x > y then x + 1 else x - 1
+  }
+  use ℤ, instMagmaInt
+  simp [Equation1648, Equation151]
+
+  constructor
+  · intro x y
+    simp [instMagmaInt]
```
Comment (pitmonticone) [inline equational_theories/ManuallyProved/Equation1648.lean:34]: Please use `simp only` here too.

File: equational_theories/ManuallyProved/Equation1648.lean [hunk 2]
```diff
@@ -19,3 +20,19 @@ theorem Equation1648_not_implies_Equation206 : ∃ (G: Type) (_: Magma G), Equat
...
+    split <;> simp <;> split <;> simp_all <;> linarith
```
Comment (pitmonticone) [inline equational_theories/ManuallyProved/Equation1648.lean:35]: Please use `simp only` here too.

File: equational_theories/ManuallyProved/Equation1648.lean [hunk 3]
```diff
@@ -19,3 +20,19 @@ theorem Equation1648_not_implies_Equation206 : ∃ (G: Type) (_: Magma G), Equat
...
+  · simp [instMagmaInt]
```
Comment (pitmonticone) [inline equational_theories/ManuallyProved/Equation1648.lean:36]: Please use `simp only` here too.

File: equational_theories/ManuallyProved/Equation1648.lean [hunk 4]
```diff
@@ -19,3 +20,19 @@ theorem Equation1648_not_implies_Equation206 : ∃ (G: Type) (_: Magma G), Equat
...
   · use 0,-1
     simp [instMagmaInt, sign]
+
+theorem Equation1648_not_implies_Equation151 : ∃ (G: Type) (_: Magma G),
+  Equation1648 G ∧ ¬ Equation151 G := by
+  let instMagmaInt : Magma ℤ := {
+    op := fun x y => if x > y then x + 1 else x - 1
+  }
+  use ℤ, instMagmaInt
+  simp [Equation1648, Equation151]
```
[the remark this answers is not in the document]
Comment (Shreyas4991) [reply equational_theories/ManuallyProved/Equation1648.lean:30]: These `simp` occurrences also apply other lemmas. It doesn't make sense to change it to `simp only`

File: equational_theories/ManuallyProved/Equation1648.lean (hunk 3 above)
Comment (Shreyas4991) [reply equational_theories/ManuallyProved/Equation1648.lean:36]: Same as above. `simp` is applying more useful simplifications here.

File: equational_theories/ManuallyProved/Equation1648.lean (hunk 2 above)
Comment (Shreyas4991) [reply equational_theories/ManuallyProved/Equation1648.lean:35]: Same as  above  : https://github.com/teorth/equational_theories/pull/569#discussion_r1799685118

File: equational_theories/ManuallyProved/Equation1648.lean (hunk 4 above)
[the remark this answers is not in the document]
Comment (pitmonticone) [reply equational_theories/ManuallyProved/Equation1648.lean:30]: I thought I should review as if we were in Mathlib. There are underlying technical reasons why `simp only` it's better. See [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/flexible.20vs.20rigid.20tactics) for further context.

**Pull request merged**